src/migrateUIFolder.ts

```diff
@@ -231,50 +257,80 @@ export async function migrateUIFolder(
-            bookmark,
-            servers,
-            keysOfWidgetPluginsToRemove,
-          );
-          dashboards[id] = migratedDashboard;
-          migratedUIFolder.children!.dashboards.children!.content.children![
-            id
-          ] = {
+          const [succesfullyMigratedDashboard, dashboardReport] =
+            migrateDashboard(bookmark, servers, keysOfWidgetPluginsToRemove);
+          dashboardMigrationReport = dashboardReport;
+          migratedDashboard = succesfullyMigratedDashboard;
+        } catch (error) {
+          fileErrorReport = createFileErrorReport(id, bookmark.name, error);
+          migratedDashboard = bookmark;
+        }
+
+        dashboards[id] = migratedDashboard;
+        migratedUIFolder.children!.dashboards.children!.content.children![id] =
+          {
             entry: {
               ...entry,
               content: JSON.stringify(_omit(migratedDashboard, ["name"])),
             },
           };
-        } catch (error) {
-          // eslint-disable-next-line no-console
-          console.error(
-            `An error occurred during the migration of dashboard ${id} ("${
-              bookmark.name
-              // Even though errors can be anything in theory, in practice they are always expected to be instances of Error.
-            }"). Ignoring this dashboard. Error:\n${(error as Error).stack}`,
+
+        if (dashboardMigrationReport || fileErrorReport) {
+          _setWith(
+            migrationReport,
+            ["dashboards", id],
+            dashboardMigrationReport || fileErrorReport,
+            Object,
           );
         }
       } else {
+        // Ignore widgets whose pluigin keys that have been flagged for removal.
+        if (
+          keysOfWidgetPluginsToRemove?.includes(
+            _getLegacyWidgetPluginKey(bookmark),
+          )
+        ) {
+          continue;
+        }
+
+        let migratedWidget = undefined;
         try {
-          if (
-            keysOfWidgetPluginsToRemove?.includes(
-              _getLegacyWidgetPluginKey(bookmark),
-            )
-          ) {
-            continue;
+          migratedWidget = migrateWidget(bookmark, servers);
+        } catch (error) {
+          if (error instanceof UnsupportedWidgetKeyError) {
+            // Migration failed because of an unknown plugin key, the widget state is migrated as-is.
+            migratedWidget = {
+              ...bookmark.value.body,
+              name: bookmark.name,
+              widgetKey: error.widgetPluginKey,
+            };
           }
 
-          const migratedWidget = migrateWidget(bookmark, servers);
+          // Whatever the error type, an error report is created.
+          const widgetErrorReport: FileErrorReport = createFileErrorReport(
+            id,
+            bookmark.name,
+            error,
+          );
+          _setWith(migrationReport, ["widgets", id], widgetErrorReport, Object);
```

Comment:
`_set` will create an array if it can interpet any part of the path as a number, even if it is a string. Using this guarantees that only objects are created, even if `id` is a stringified number.